docs/setup.md

```diff
@@ -0,0 +1,88 @@
+3. Disable the _'Start Docker when you log in'_ setting under the 'General' tab.
+
+### Lando
+
+This project supports development with Lando using the Pantheon recipe. This produces a local development environment that matches the hardware and configuration of the Pantheon platform. Most local tooling, including Composer, Drush, Drupal Console, and node applications are available through this recipe.
+
+1. [Review the preflight checks](https://docs.devwithlando.io/installation/preflight.html)
+2. [Download and install the latest release](https://github.com/lando/lando/releases)
+3. [Optional: Setup local certificate authority](https://docs.devwithlando.io/config/security.html)
+
+### Additional tools
+
+* [Composer](https://getcomposer.org/download/): PHP package manager. Version 2.x.
+* [NVM](https://github.com/nvm-sh/nvm#install--update-script): Recommended for installing and switching between Node versions.
+* Node.js (>=8.0). We recommend installing via NVM.
+
+## Project Setup
+
+This repository contains a custom Pantheon Upstream used to create and manage every site on the YaleSites platform. Out of the box this project is not connected to an individual Drupal site. To contribute to this project, we need to connect the local development environment to a Drupal site to leverage the site's files and database.
+
+```bash
+# Step 1: Clone this repository and enter the project directory.
+git clone https://github.com/yalesites-org/yalesites-project.git
+cd yalesites-project
+
+# Step 2: Ignore the composer.lock file on local dev only.
+grep -qxF 'composer.lock' .git/info/exclude || echo 'composer.lock' >> .git/info/exclude
+```
+
+The Pantheon Lando recipe can connect the local development environment to a remotely hosted site. Connection information is stored in a local Lando settings file and includes a Pantheon site name and UUID. While we can connect to any site running this upstream, a particular integration environment has been provisioned for engineers working on this repository or the associated YaleSites profile. Connection information for the integration environment is stored in the example Lando local file.
+
+```bash
+# Step 3: Create a local Lando settings file to connect to the external environment.
+cp .lando.local.example.yml .lando.local.yml
+```
+
+Starting Lando will provision the containers required to run a local development environment. The `pull` command can then sync the database and files from a Pantheon-hosted site with the new local site.
+
+```bash
+# Step 4: Start Lando and import the remote files and database.
+lando start
+lando pull --database=dev --files=dev --code=none
+```
+
+Visit the local dev site [https://yalesites-project.lndo.site/](https://yalesites-project.lndo.site/) or run `lando drush uli` to obtain a login link.
```

Comment:
When I visited the URL generated from `lando drush uli`, I got a 500 error unless I run `lando drush cr` after the `lando pull` above. Here is the output from `lando logs` about the error:

```
appserver_nginx_1  | 192.168.240.6 - - [04/Apr/2022:19:22:19 +0000] "GET /user/reset/1/1649100124/pnGgvjC3EuHXvMEwp0JIzcLvNsOuDLY9ENBsNVAgeu8/login HTTP/1.1" 500 85 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.84 Safari/537.36"
appserver_nginx_1  | 192.168.240.6 - - [04/Apr/2022:19:22:19 +0000] "GET /favicon.ico HTTP/1.1" 404 185 "https://yalesites-project.lndo.site/user/reset/1/1649100124/pnGgvjC3EuHXvMEwp0JIzcLvNsOuDLY9ENBsNVAgeu8/login" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.84 Safari/537.36"
appserver_nginx_1  | 2022/04/04 19:22:20 [error] 245#0: *13 FastCGI sent in stderr: "PHP message: Error: __clone method called on non-object in /app/web/core/lib/Drupal/Core/Routing/UrlGenerator.php on line 426 #0 /app/web/core/lib/Drupal/Core/Routing/UrlGenerator.php(270): Drupal\Core\Routing\UrlGenerator->getRoute('<current>')
appserver_nginx_1  | #1 /app/web/core/lib/Drupal/Core/Render/MetadataBubblingUrlGenerator.php(105): Drupal\Core\Routing\UrlGenerator->generateFromRoute('<current>', Array, Array, true)
appserver_nginx_1  | #2 /app/web/modules/contrib/redirect/src/EventSubscriber/RouteNormalizerRequestSubscriber.php(105): Drupal\Core\Render\MetadataBubblingUrlGenerator->generateFromRoute('<current>', Array, Array)
appserver_nginx_1  | #3 [internal function]: Drupal\redirect\EventSubscriber\RouteNormalizerRequestSubscriber->onKernelRequestRedirect(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request', Object(Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher))
appserver_nginx_1  | #4 /app/web/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php(142): call_user_func(Array, Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request', Object(Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher))
appserver_nginx_1  | #5 /app/vendor/symfony/http-kernel/HttpKernel.php(134): Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request')
appserver_nginx_1  | #6 /app/vendor/symfony/http-kernel/HttpKernel.php(80): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
appserver_nginx_1  | #7 /app/web/core/lib/Drupal/Core/StackMiddleware/Session.php(58): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
appserver_nginx_1  | #8 /app/web/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(48): Drupal\Core\StackMiddleware\Session->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
appserver_nginx_1  | #9 /app/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(191): Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object(Symfony
```

Reply:
Clear the cache first - always the story with Drupal. I'm wondering if this is a transient error. In any case I added a cache clear step to the documentation to be safe. 

I'm now thinking we should create a npm script to execute all of these steps in one command. `npm run setup` or something like that.

Reply:
Yes, not sure - I did `lando destroy` multiple times when testing these docs and had to clear caches every time after pulling. Definitely could combine them into a single script.